### DISCUSSION TITLE
Ergonomic UI for anchored-links

### DISCUSSION THREAD

**Discussion by daladim** [outline/outline#3562] [Ideas]
That's a follow-up to https://github.com/outline/outline/issues/3553

It looks you already did a good job supporting anchored-links (cf https://github.com/outline/outline/issues/739#issuecomment-410547781), that's a term I've just made up to describe "a link to a specific sub-section in a page".

But this could be made even better by supporting them as first-class links by:
* having a way in the UI to create such links. Currently, creating a link (ctrl-K) prompts for a link, and autocompletes page names in the current wiki (which is great). But it may also provide the ability to point to a specific anchor when it detects it points at the current wiki.
* such links to a sub-section should be automatically "fixed" when the sub-section is renamed, the same way links to pages are "fixed" when the page is renamed

I'm sorry for not being a webdev, and I'm afraid I won't be able to fix this myself

Thanks!

**Comment by Limezy** [comment]
Fully agreed with everything above.
I would even go further, for two things : 
1. Hover with the mouse on top of such links should show a preview of the header linked to, not beginning of the doc
2. Would be nice to be able to link any item in the editor (paragraph, list, integration component etc...)
